src/libmongoc/tests/TestSuite.c

```diff
@@ -1056,6 +1193,20 @@ TestSuite_Destroy (TestSuite *suite)
    }
 
    _mongoc_array_destroy (&suite->match_patterns);
+
+   for (i = 0; i < suite->failing_flaky_skips.len; i++) {
+      TestSkip *val =
+         _mongoc_array_index (&suite->failing_flaky_skips, TestSkip *, i);
+      if (val->test_name)
+         bson_free (val->test_name);
+      if (val->subtest_desc)
+         bson_free (val->subtest_desc);
+      if (val->reason)
+         bson_free (val->reason);
```

Comment:
```suggestion
      bson_free (val->test_name);
      bson_free (val->subtest_desc);
      bson_free (val->reason);
```
Calls to `bson_free` do not require a null check.

Reply:
That's interesting.  The `if` statements are clearly unnecessary, but for some reason at the time I implemented that section it didn't seem that way to me.  Thanks for catching it.